src/components/pager/pager.test.js

```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
+          <PagerComponent showTotalRecords={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "should render Pager with showFirstAndLastButtons prop set to %s for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent showFirstAndLastButtons={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "should render Pager with showPreviousAndNextButtons prop set to %s for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent showPreviousAndNextButtons={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "should render Pager with showPageCount prop set to %s for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(<PagerComponent showPageCount={boolVal} />);
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(["default", "alternate"])(
+      "should render Pager with variant prop set to %s for accessibility tests",
+      (variant) => {
+        CypressMountWithProviders(<PagerComponent variant={variant} />);
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "pager links are rendered as intended when hideDisabledElements is set to %s and currentPage is 1 for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent currentPage={1} hideDisabledElements={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([true, false])(
+      "pager links are rendered as intended when hideDisabledElements is set to %s and currentPage is 10 for accessibility tests",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PagerComponent currentPage={10} hideDisabledElements={boolVal} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    it("both pager links are rendered when hideDisabledElements is set to true, but currenPage is above 1 for accessibility tests", () => {
+      CypressMountWithProviders(
+        <PagerComponent currentPage={7} hideDisabledElements />
+      );
+      cy.checkAccessibility();
+    });
+
+    describe("check funtionality for Pager component for accessibility tests", () => {
+      it.each([-1, -10, -100, testData[0], testData[1]])(
+        "should set totalRecords out of scope to %s for accessibility tests",
+        (totalRecords) => {
+          CypressMountWithProviders(
+            <PagerComponent totalRecords={totalRecords} />
+          );
+          cy.checkAccessibility();
+        }
+      );
+
+      it("should disable nextArrow and lastArrow buttons after clicking on lastArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={3} />);
+        cy.checkAccessibility();
```

Comment:
This test doesn't actually disable anything as nothing is clicked.

Reply:
it shouldn't :)